extended condition not to add the content-length header negative

kjezek opened the pull request:
This update prevents wrong adding of the Content-Length header with the value -1. Such a value is not defined in the specification. The header should not be present at all when the length is not defined.

kjezek commented:
This should fix: https://github.com/eclipse-ee4j/jersey/issues/4965

jansupol commented:
Thank you for the PR. The continuous integration check did fail because the copyright year is not updated to 2022.

The PR does not seem right. Why did you exclude zero-length content? Can you provide a test that demonstrates the use-case this PR helps?

kjezek commented:
Hi @jansupol 
1. I have fixed the license year issues.

2. The Content-Length header must not be added to the header list when the length is negative. First, it is against the specification - when the length is not know, the header must be omitted, not used with a negative number.  Secondly, the code I have fixed did originally add the wrong Content-Length header when the header was actually not present in the original response - i.e. it modified the response wrongly, adding a wrong header.

jansupol commented:
I can understand the negative number.
My question was why did you want to remove header for the zero length content, i.e. when entity.getContentLength() == 0?

kjezek commented:
@jansupol you are right, the length of Content-Length is a valid value.  I have fixed the condition.
